apps/librelingo_yaml_loader/tests/test_yaml_loader_convert_phrase.py

```diff
@@ -0,0 +1,82 @@
+"""
+this file contains tests of the funcion
+librelingo_yaml_loader.yaml_loader._convert_phrase
+"""
+import pytest
+
+from librelingo_types import Phrase
+from librelingo_yaml_loader.yaml_loader import _convert_phrase
+
+from librelingo_fakes import fakes
+
+
+def _alt_versions():
+    return "Alternative versions"
+
+
+def _alt_translations():
+    return "Alternative translations"
+
+
+@pytest.fixture
+def raw_fake_phrase():
+    """returns the raw data describing the phrase used in tests in this file"""
+    return {
+        "Phrase": fakes.fake_value(),
+        _alt_versions(): [
+            fakes.fake_value(),
+            fakes.fake_value(),
+        ],
+        "Translation": fakes.fake_value(),
+        _alt_translations(): [
+            fakes.fake_value(),
+            fakes.fake_value(),
+        ],
+    }
+
+
+def test_returns_a_phrase_object(raw_fake_phrase):
+    assert isinstance(_convert_phrase(raw_fake_phrase), Phrase)
+
+
+def test_includes_main_version(raw_fake_phrase):
+    assert (
+        _convert_phrase(raw_fake_phrase).in_target_language[0]
+        == raw_fake_phrase["Phrase"]
+    )
+
+
+def test_includes_alternative_versions(raw_fake_phrase):
+    result = _convert_phrase(raw_fake_phrase).in_target_language
+    assert raw_fake_phrase[_alt_versions()][0] in result
```

Comment:
this is a weird usage, why not just use the string :thinking: or is there some anti-duplication rule that enforces this (pylint?) :thinking:

Reply:
Sonar was complaining about it (some "code smell").

Reply:
why is it a code smell and is there a different way to deal with it?

Reply:
The complain was the string `"Alternative versions"` was repeated more then 3 (?, I am not sure) times. Similarly with `"Alternative translations"`.

Why is this a code smell? I think it is quite clear why to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). In this specific case I would also argue that avoiding duplication reduces the probability of making a typo in future tests.

Is there a different way? Well:
- simply define a _constant_ `_ALT_VERSIONS = "Alternative versions"` (but people are more keen to overwrite a string than a function),
- define a function 
```python
def get_alternative_versions(in_raw_data):
    return in_raw_data["Alternative versions"]
```
and use it when needed. One could argue here that such approach separates the test from the structure of the raw data. But in principle one should also write tests for the function `get_alternative_versions()`.

Do you know another way (besides removing few of the problematic tests)?

Reply:
> simply define a constant _ALT_VERSIONS = "Alternative versions" (but people are more keen to overwrite a string than a function),

actually I don't think this reduces the chance of a typo. you will get a different error but in most examples you should get an error nevertheless 🤔 

I can see the "define a function" idea that you are citing is the same I just wrote! I think that's a good idea.

> Do you know another way (besides removing few of the problematic tests)?

Perhaps instead of random fake  values you can use some concrete example, then

```python
assert raw_fake_phrase[_alt_versions()][0] in result
```

maybe be replaced with

```python
assert "She sells sea shells" in result
```